auditing/auditing.go

```diff
@@ -0,0 +1,97 @@
+package auditing
+
+import (
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type Config struct {
+	Component        string
+	URL              string
+	APIKey           string
+	IndexPrefix      string
+	RotationInterval Interval
+	Keep             int64
+	Log              *zap.SugaredLogger
+}
+
+type Interval string
+
+var (
+	HourlyInterval  Interval = "@hourly"
+	DailyInterval   Interval = "@daily"
+	MonthlyInterval Interval = "@monthly"
+)
+
+type EntryType string
+
+const (
+	EntryTypeHTTP  EntryType = "http"
+	EntryTypeGRPC  EntryType = "grpc"
+	EntryTypeEvent EntryType = "event"
+)
+
+type EntryDetail string
+
+const (
+	EntryDetailGRPCUnary  EntryDetail = "unary"
+	EntryDetailGRPCStream EntryDetail = "stream"
+)
+
+type EntryPhase string
+
+const (
+	EntryPhaseRequest  = "request"
```

Comment:
these const are not typed

Reply:
Good catch! Inserted type for all consts :+1: